app/views/questionnaire.py

```diff
@@ -121,6 +121,10 @@ def post_block(eq_id, form_type, collection_id, group_id, group_instance, block_
     else:
         _update_questionnaire_store(current_location, form)
         next_location = path_finder.get_next_location(current_location=current_location)
+
+        if next_location is None:
```

Comment:
What if the last page in the survey is an additional interstitial and not the summary or final-confimation?